src/popup/router/pages/PermissionsDetails.vue

```diff
@@ -0,0 +1,169 @@
+    </div>
+
+    <div class="permission-row">
+      <CheckBox :value="messageSign" @input="togglePermission({ host, name: 'messageSign' })" />
+      <span :class="{ highlight: messageSign }"> {{ $t('pages.permissions.message-sign') }}</span>
+    </div>
+
+    <div class="transaction-sign-limit">
+      <div class="permission-row">
+        {{ $t('pages.permissions.transaction-sign') }}
+        <Input
+          :value="transactionSignLimit || ''"
+          :error="transactionSignLimitError"
+          placeholder="no limit"
+          @input="setTransactionSignLimit"
+        />
+      </div>
+      <RangeInput
+        :value="transactionSignLimit"
+        min="0"
+        :max="tokenBalance"
+        step="0.1"
+        @input="setTransactionSignLimit"
+      />
+      <div class="permission-row">
+        {{ $t('pages.permissions.spent-today') }}
+        <TokenAmount :amount="transactionSignLimit - transactionSignLimitLeft" />
+      </div>
+      <div class="permission-row">
+        {{ $t('pages.permissions.left-today') }}
+        <TokenAmount :amount="transactionSignLimitLeft" />
+      </div>
+      <div class="permission-row">
+        {{ $t('pages.account.balance') }}
+        <TokenAmount :amount="tokenBalance" />
+      </div>
+    </div>
+    <Button bold @click="removePermissions">
+      {{ $t('pages.permissions.delete') }}
+    </Button>
+  </div>
+</template>
+
+<script>
+import { pick } from 'lodash-es';
+import { mapState, mapMutations } from 'vuex';
+import CheckBox from '../components/CheckBox';
+import Input from '../components/Input';
+import RangeInput from '../components/RangeInput';
+import TokenAmount from '../components/TokenAmount';
+import Button from '../components/Button';
+
+export default {
+  components: {
+    CheckBox,
+    Input,
+    RangeInput,
+    TokenAmount,
+    Button,
+  },
+  data: () => ({
+    transactionSignLimitError: false,
+  }),
+  subscriptions() {
+    return pick(this.$store.state.observables, ['tokenBalance']);
+  },
+  computed: {
+    host() {
+      return this.$route.params.host;
+    },
+    ...mapState(
+      'permissions',
+      ['address', 'messageSign', 'transactionSignLimit', 'transactionSignLimitLeft'].reduce(
+        (c, s) => ({
+          ...c,
+          [s](state) {
+            return state[this.host][s];
+          },
+        }),
+        {},
+      ),
+    ),
+  },
+  methods: {
+    ...mapMutations('permissions', ['togglePermission']),
+    setTransactionSignLimit(rawValue) {
+      const value = +rawValue;
+      this.transactionSignLimitError = Number.isNaN(value) || value < 0;
+      if (this.transactionSignLimitError) return;
+      this.$store.commit('permissions/setTransactionSignLimit', { host: this.host, value });
+    },
+    removePermissions() {
+      this.$store.commit('permissions/removeAeppPermissions', this.host);
+      this.$router.replace({ name: 'not-found' });
```

Comment:
I got why it is needed, but won't be better to show 404 when navigating to an unexisting permission page like https://feature-applications-permissions.wallet.z52da5wt.xyz/settings/permissions/asdflkjhkljhk? It may be done by redirection to the 404 page in `created` hook if `host` does not exist in the permissions